heat_infoblox/resources/bgp.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2016 Infoblox Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import logging
+
+from heat.common.i18n import _
+from heat.engine import constraints
+from heat.engine import properties
+from heat.engine import resource
+from heat.engine import support
+from oslo_concurrency import lockutils
+
+from heat_infoblox import constants
+from heat_infoblox import resource_utils
+
+LOG = logging.getLogger(__name__)
+
+
+class Bgp(resource.Resource):
+    """A resource which represents a BGP Neighbor configuration.
+
+    This resource is used configure a BGP Autonomous System and single BGP
+    Neighbor per resource.
```

Comment:
Multiple neighbors can exist for one AS. And this resource contains data for both AS and Neighbor.

Currently it works next way:

A. You have two neighbor resources with equal AS data.
When the first BGP resource is created it generates AS section and one neighbor.
When the second BGP resource is created AS section is not touched and one new neighbor is appended to neighbors list.

B. You have two neighbor resources with different AS data.
When the first BGP resource is created it generates AS section and one neighbor.
When the second BGP resource is created AS section gets updated with AS data from the second member and one new neighbor is appended to neighbors list.

For case B we have a little bit mess, because AS is changed during creation of the second template.

If that doesn't work for us I can split this BGP resource into two resources: one for BGP AS and one for BGP Neighbor. In this case configuring the same BGP scenario with two neighbors will require three stacks built from two base templates:
- BGP AS stack (BGP as base template)
- BGP Neighbor 1 stack (BGP neighbor base template)
- BGP Neighbor 2 stack (BGP neighbor base template)
Let me know if this way looks better.